RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -617,12 +617,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
                     outputGPU.chi2()[gch] = -9999.f;
 
                   // check as in cpu version if mahi is not needed
-                  // FIXME: KNOWN ISSUE: observed a problem when rawCharge and pedestal
-                  // are basically equal and generate -0.00000...
-                  // needs to be treated properly
-                  if (!(shrEnergyM0TotalAccum[lch] > 0 && energym0_per_ts_gain0 > ts4Thresh)) {
-                    // do not need to run mahi minimization
-                    //outputEnergy[gch] = 0; energy already inited to 0
+                  if (energym0_per_ts_gain0 <= ts4Thresh) {
```

Comment:
could you change this to
```suggestion
                  if (not energym0_per_ts_gain0 > ts4Thresh) {
```
?

This way if `energym0_per_ts_gain0` is a NaN, the comparison will be false, the `not` will make it true, and the channel will be marked as invalid.

Reply:
Thanks for the comment, I pushed the change (I added parentheses, to make the logic more obvious.. to me).

Reply:
In the latest push (ea1a1bd0b23d5225d2153b8587cfed612bd48c87), I did the same for the 2nd check in this PR (as suggested offline by Andrea).

Reply:
It might be useful to add a comment for the reason using `not .. >` instead of `<=` to reduce the probability this would get (inadvertently) changed in the future.